.github/workflows/update-maintainers.yml

```
@@ -0,0 +1,245 @@
+        with:
+          github_access_token: ${{ steps.app-token.outputs.token || github.token }}
+          extra_nix_config: |
+            allow-import-from-derivation = false
+
+      - name: Setup Git
+        env:
+          name: ${{ steps.user-info.outputs.name || 'github-actions[bot]' }}
+          email: ${{ steps.user-info.outputs.email || '41898282+github-actions[bot]@users.noreply.github.com' }}
+        run: |
+          git config user.name "$name"
+          git config user.email "$email"
+
+      - name: Generate updated maintainers list
+        run: |
+          echo "::group::📋 Generating updated generated/all-maintainers.nix..."
+          nix run .#generate-all-maintainers -- --root . --output generated/all-maintainers.nix
+          echo "::endgroup::"
+          echo "::group::🎨 Formatting with nixfmt..."
+          nix fmt generated/all-maintainers.nix
+          echo "::endgroup::"
+
+      - name: Check for changes
+        id: check-changes
+        run: |
+          if git diff --quiet generated/all-maintainers.nix; then
+            echo "No changes to generated/all-maintainers.nix"
+            echo "has_changes=false" >> "$GITHUB_OUTPUT"
+          else
+            echo "Changes detected in generated/all-maintainers.nix"
+            echo "has_changes=true" >> "$GITHUB_OUTPUT"
+            added=$(git diff --numstat generated/all-maintainers.nix | cut -f1)
+            removed=$(git diff --numstat generated/all-maintainers.nix | cut -f2)
+            echo "changes_summary=+$added -$removed lines" >> "$GITHUB_OUTPUT"
+          fi
+
+      - name: Validate generated file
+        if: steps.check-changes.outputs.has_changes == 'true'
+        run: |
+          echo "🔍 Validating generated generated/all-maintainers.nix..."
+          if nix-instantiate --eval  generated/all-maintainers.nix --strict > /dev/null; then
+            echo "✅ Generated file has valid Nix syntax"
+          else
+            echo "❌ Generated file has invalid Nix syntax"
+            exit 1
+          fi
+
+      - name: Create update branch
+        run: |
+          git branch -D "$pr_branch" || echo "Nothing to delete"
+          git switch -c "$pr_branch"
+
+      - name: Get info on the current PR
+        id: open_pr_info
+        env:
+          GH_TOKEN: ${{ steps.app-token.outputs.token || github.token }}
+        run: |
+          # Query for info about the already open update PR
+          info=$(
+            gh api graphql -F owner='{owner}' -F repo='{repo}' -F branch="$pr_branch" -f query='
+              query($owner:String!, $repo:String!, $branch:String!) {
+                repository(owner: $owner, name: $repo) {
+                  pullRequests(first: 1, states: OPEN, headRefName: $branch) {
+                    nodes {
+                      number
+                      url
+                    }
+                  }
+                }
+              }
+            ' | jq --raw-output '
```

FYI `gh api` also has a `--jq` flag:
```suggestion
            ' --jq '
```

Sometimes it's still worth it to pipe to jq though, e.g. if you want to use `--exit-status` to trigger a failure if the jq query doesn't match the input JSON.


---


We use a similar graphql query elsewhere, but one issue I'm now spotting is that `headRefName` could also match PRs on forks that use the same branch name... I don't see an obvious quick fix in the graphql explorer, and the issue applies to both places, so this is fine for now.